File: C++/construct_the_rectangle.cpp
class Solution {
public:
    vector<int> constructRectangle(int area) {
        vector<int> result;
        int n = int(sqrt(area));
        
        for(int i = n; n > 0; i--) {
            if(area % i == 0) {
                result.push_back(i);
                result.push_back(area / i);
                break;
            }
        }
                
        if(result[0] < result[1]) {
            int tmp = result[0];
            result[0] = result[1];
            result[1] = tmp;
        }
            
        return result;
        
    }
};

File: Python3/construct_the_rectangle.py
class Solution:
    def constructRectangle(self, area: int) -> List[int]:
        result = []
        n = int(sqrt(area))
        
        for i in range(n, 0, -1):
            if area % i == 0:
                result.append(i)
                result.append(area // i)
                break
                
        if result[0] < result[1]:
            tmp = result[0]
            result[0] = result[1]
            result[1] = tmp
            
        return result
